src/modules/site-v2/base/views/data/data.py

```diff
@@ -57,3 +76,186 @@ def protocols():
     }
     return render_template('data/protocols.html', **params)
 
+
+#
+# Submit Trait
+#
+@data_bp.route('/submit-trait')
```

Comment:
For the endpoints, let's normalize around one of the standards, eg: REST:
- /content/traits [GET] list all the traits
- /content/traits [POST] create a new trait
- /content/trait/:id [GET] View single trait
- /content/trait/:id [PUT] Modify single trait
- /content/trait/:id [DELETE] Delete single trait.

Since some of these operations will require an edit/modify view or delete confirmation, consider also these:
- /content/traits/create [GET] form for new trait creation. The form posts to /content/traits [POST]
- /content/traits/:id/edit [GET] form to edit the trait. The form 'puts' to /content/traits/:id [PUT] 
- /content/traits/:id/delete [GET] form to confirm the deletion of the trait. The form posts to /content/traits/:id [DELETE]

Reply:
Thanks for touching the base on this. Just to clarify -- do we need to keep all traits endpoints under the data, like `/data/trait(s)/*` or you think they should under separate route (like you proposed "/content"?

Reply:
sry, not at all, keeping them under `/data/*` is perfectly fine.

Reply:
Pushed in the commit c3b030566d24eb320d154dac1f8c1f418ad61c95